### PR TITLE
date: fix error message when opening file

### PR DESCRIPTION
**Pull request by leon3s**
Closes https://github.com/uutils/coreutils/issues/4540

### CODE REVIEW

File: tests/by-util/test_date.rs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 extern crate regex;
 
+use std::os::unix::prelude::PermissionsExt;
```
Comment (sylvestre) [inline tests/by-util/test_date.rs:3]: I think you need to make that line Unix specific - will fail on Windows

**Comment by leon3s** [comment]
I cannot really test the behavior on window so maybe i enable the test on unix only ?

**Comment by sylvestre** [comment]
yeah, unix only is fine

File: src/uu/date/src/date.rs [hunk 2]
```diff
@@ -225,10 +225,10 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                     let iter = lines.filter_map(Result::ok).map(parse_date);
                     Box::new(iter)
                 }
-                Err(_err) => {
+                Err(err) => {
                     return Err(USimpleError::new(
                         2,
-                        format!("{}: No such file or directory", path.display()),
+                        format!("{}: {}", path.display(), err.kind()),
                     ));
```
Comment (tertsdiepraam) [inline src/uu/date/src/date.rs:232]: Seems like a good use case for [`map_err_context`](https://docs.rs/uucore/latest/uucore/error/trait.FromIo.html#tymethod.map_err_context), which can turn an `std::io::Error` into a `UError`.

**Comment by sylvestre** [comment]
Fails on mac:
```

    Checking uu_du v0.0.17 (/Users/runner/work/coreutils/coreutils/src/uu/du)
error[E0599]: no method named `map_err_context` found for struct `Error` in the current scope
   --> src/uu/date/src/date.rs:229:36
    |
229 |                     return Err(err.map_err_context(|| path.display().to_string()));
    |                                    ^^^^^^^^^^^^^^^ method not found in `Error`
    |
   ::: /Users/runner/work/coreutils/coreutils/src/uucore/src/lib/mods/error.rs:468:8
    |
468 |     fn map_err_context(self, context: impl FnOnce() -> String) -> T;
    |        --------------- the method is available for `std::io::Error` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
11  | use uucore::error::FromIo;
    |

For more information about this error, try `rustc --explain E0599`.
```

**Comment by leon3s** [comment]
It's look like the failing tests aren't related to the pull request itself now

**Comment by tertsdiepraam** [comment]
I just realized we have some duplicate work going on between this PR and https://github.com/uutils/coreutils/pull/4572. I think the latter is a bit more comprehensive. But maybe we can add the test from this PR to the other one?

**Comment by leon3s** [comment]
So i draft this one maybe and when https://github.com/uutils/coreutils/pull/4572 is ready i add the test to it ?

**Comment by tertsdiepraam** [comment]
Sounds like a good plan :)

**Comment by cakebaker** [comment]
> So i draft this one maybe and when https://github.com/uutils/coreutils/pull/4572 is ready i add the test to it ?

@leon3s just fyi, #4572 has been merged a while ago

**Pull request merged**